src/m365/planner/commands/task/task-add.ts

```diff
@@ -0,0 +1,387 @@
+    telemetryProps.title = typeof args.options.title !== 'undefined';
+    telemetryProps.startDateTime = typeof args.options.startDateTime !== 'undefined';
+    telemetryProps.dueDateTime = typeof args.options.dueDateTime !== 'undefined';
+    telemetryProps.percentComplete = typeof args.options.percentComplete !== 'undefined';
+    telemetryProps.assignedToUserIds = typeof args.options.assignedToUserIds !== 'undefined';
+    telemetryProps.assignedToUserNames = typeof args.options.assignedToUserNames !== 'undefined';
+    telemetryProps.description = typeof args.options.description !== 'undefined';
+    telemetryProps.orderHint = typeof args.options.orderHint !== 'undefined';
+    return telemetryProps;
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['title', 'planId', 'bucketId'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this
+      .getPlanId(args)
+      .then((planId: string) => 
+        this.getBucketId(args, planId).then(async (bucketId: string) => {
+          const assignments: { [userId: string]: any; } = await this.generateUserAssignments(args, logger);
+
+          const requestOptions: any = {
+            url: `${this.resource}/v1.0/planner/tasks`,
+            headers: {
+              'accept': 'application/json;odata.metadata=none'
+            },
+            responseType: 'json',
+            data: {
+              planId: planId,
+              bucketId: bucketId,
+              title: args.options.title,
+              startDateTime: args.options.startDateTime,
+              dueDateTime: args.options.dueDateTime,
+              percentComplete: args.options.percentComplete,
+              assignments: assignments,
+              orderHint: args.options.orderHint
+            }
+          };
+
+          return request
+            .post(requestOptions)
+            .then(async (newTask: any) => {
+              const taskId = newTask['id'];
+
+              if (args.options.description) {
+                const etag = await this.getTaskDetails(taskId);
+
+                const requestOptionsTaskDetails: any = {
+                  url: `${this.resource}/v1.0/planner/tasks/${taskId}/details`,
+                  headers: {
+                    'accept': 'application/json;odata.metadata=none',
+                    'If-Match': etag
+                  },
+                  responseType: 'json',
+                  data: {
+                    description: args.options.description
+                  }
+                };
+
+                await request.patch(requestOptionsTaskDetails);
+
+                return newTask;
+              } 
+
+              return newTask;
+            });
+        })
+      )
+      .then((res: any): void => {
+        logger.log(res);
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getTaskDetails(taskId: string): Promise<any> {
```

Comment:
Since we know the type of data returned by the API, let's use the exact type rather than `any`

Reply:
Problem with the TaskDetails is that I need that etag from odata. When is use the _PlannerTaskDetails_ model from `@microsoft/microsoft-graph-types` all my odata properties are gonne. So currently don't really have a work around the `any` type.

Reply:
Seems like we're resolving the promise with the task's etag which is a string. In that case, I'd suggest changing the return type to `Promise<string>` and renaming the method to `getTaskEtag` to clarify its purpose

Reply:
Yeah, that's a good idea. I will change it